### PR TITLE
Predict two-way levers

### PR DESCRIPTION
**Pull request by metalgearsloth**
Annoys me the rare occasions I touch cargo. Doesn't predict the signal but at least the lever responds immediately.

### CODE REVIEW

**Comment by Jezithyr** [comment]
LGTM I don't see any non-nitpick issues with this. Might be worth using Entity\<T?\> next time instead of the old EntityUid, Comp? way of implementing the resolve pattern.

**Review by 0x6273 — CHANGES_REQUESTED** [review]
There are "unnecessary `using` directives" warnings in some files after you moved stuff to shared namespace.

File: Content.Shared/DeviceLinking/SharedDeviceLinkSystem.cs [hunk 1]
```diff
@@ -550,4 +557,20 @@ private void CreateLinkPopup(EntityUid? userId, EntityUid sourceUid, string sour
...
+    /// <summary>
+    /// Sends a network payload directed at the sink entity.
+    /// Just raises a <see cref="SignalReceivedEvent"/> without data if the source or the sink doesn't have a <see cref="DeviceNetworkComponent"/>
+    /// </summary>
+    /// <param name="uid">The source uid that invokes the port</param>
+    /// <param name="port">The port to invoke</param>
+    /// <param name="data">Optional data to send along</param>
+    /// <param name="sourceComponent"></param>
+    public virtual void InvokePort(EntityUid uid, string port, NetworkPayload? data = null,
+        DeviceLinkSourceComponent? sourceComponent = null)
```
[the remark this answers is not in the document]
Comment (Jezithyr) [reply Content.Shared/DeviceLinking/SharedDeviceLinkSystem.cs:571]: That's not really long enough to warrant switching it to a list imo.

**Comment by metalgearsloth** [comment]
> There are "unnecessary `using` directives" warnings in some files after you moved stuff to shared namespace.

IDE warnings are way down the list compared to compiler warnings and deleting tends to attract more merge conflicts.

**Pull request merged**